### PR TITLE
Only make new comments when we golint.

### PR DESCRIPTION
**Pull request by spxtr**
This plugin is still experimental. If this works out then it's time to switch it to listen to PR events and comment automatically, not just when someone says `/lint`.

### CODE REVIEW

File: prow/github/client.go [hunk 1]
```diff
@@ -389,6 +389,40 @@ func (c *Client) GetPullRequestChanges(org, repo string, number int) ([]PullRequ
 	return changes, nil
 }
 
+// ListPullRequestComments returns all comments on an issue. This may use more
```
Comment (cjwagner) [inline prow/github/client.go]: s/an issue/a pull request/

File: prow/plugins/golint/golint.go [hunk 2]
```diff
@@ -133,30 +161,62 @@ func handle(ghc githubClient, gc *git.Client, log *logrus.Entry, ic github.Issue
...
-				Position: patchPath,
-				Body:     problem.Text,
+				Path:     f,
+				Position: l,
+				Body:     body,
 			})
 		}
 	}
 
+	numOldProblems := numProblems(problems)
```
Comment (cjwagner) [inline prow/plugins/golint/golint.go]: Nit: This is really the total number of problems, not the number of old problems.

File: prow/github/client.go [hunk 3]
```diff
@@ -389,6 +389,40 @@ func (c *Client) GetPullRequestChanges(org, repo string, number int) ([]PullRequ
...
+// than one API token.
+func (c *Client) ListPullRequestComments(org, repo string, number int) ([]ReviewComment, error) {
```
Comment (cjwagner) [inline prow/github/client.go]: You could add a test for this function.

File: prow/github/client.go (hunk 1 above)
Comment (spxtr) [inline prow/github/client.go]: Done.

File: prow/github/client.go (hunk 3 above)
Comment (spxtr) [inline prow/github/client.go]: Done.

File: prow/plugins/golint/golint.go (hunk 2 above)
Comment (spxtr) [inline prow/plugins/golint/golint.go]: Done.

**Pull request merged**